src/main/kotlin/org/rust/ide/actions/RsAbstractMover.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions
+
+import com.intellij.codeInsight.editorActions.moveUpDown.LineRange
+import com.intellij.codeInsight.editorActions.moveUpDown.StatementUpDownMover
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.tree.IElementType
+import org.rust.ide.utils.findElementAtIgnoreWhitespaceAfter
+import org.rust.ide.utils.findElementAtIgnoreWhitespaceBefore
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.elementType
+
+val PsiElement.line: Int? get() = containingFile.viewProvider.document?.getLineNumber(textRange.startOffset)
+
+abstract class RsAbstractMover : StatementUpDownMover() {
```

Comment:
Let's name it `RsAbstractUpDownMover`, to make it easier to discover via <kbd>Ctrl+N</kbd>? The descendants should probably have `UpDown` in their names as well.